ghcide/src/Development/IDE/Core/OfInterest.hs

```diff
@@ -43,45 +45,59 @@ instance IsIdeGlobal OfInterestVar
 ofInterestRules :: Rules ()
 ofInterestRules = do
     addIdeGlobal . OfInterestVar =<< liftIO (newVar HashMap.empty)
-    defineEarlyCutOffNoFile $ \GetFilesOfInterest -> do
+    defineEarlyCutoff $ RuleNoDiagnostics $ \IsFileOfInterest f -> do
```

Comment:
Could we perhaps have both `IsFileOfInterest` (early cutoff, not always rerun) and `GetFilesOfInterest`(early cutoff, always rerun). Otherwise would could have subtle inconsistencies due to the file of interest set changing.

Reply:
The only genuine use case for `GetFilesOfInterest` is early cutoff inside a rule definition. I looked around and that use case doesn't arise anywhere. All the calls to `getFilesOfInterest` are either in handlers or in `action` clauses. 

Given that `GetFilesOfInterest` is strictly more general than `IsFileOfInterest` but has catastrophic performance, I think it's best not to include it. The inconsistency that you describe cannot arise realistically since whenever the set is mutated, the Shake session is restarted.